docs/proposals/advanced-nginx-extensions.md

```diff
@@ -26,3 +74,915 @@ NGINX use cases. To allow them to implement those use cases, we need to bring a
 
 - Support configuration other than NGINX directives. For example, njs configuration files or TLS certificates.
 - Reimplement already supported features through the new extension mechanism.
+- Allow users to customize supported NGINX configuration (for example, add a parameter to an NGINX directive).
+
+## Advanced Extensions
+
+This proposal brings two extension mechanisms:
+
+- [SnippetsPolicy](#snippetspolicy) which allows to quickly bring unsupported NGINX configuration to NGF. However,
+  because of its implications for reliability and security, SnippetsPolicy is mostly applicable for Cluster operator.
+- [SnippetsTemplate](#snippetstemplate) which allows to bring unsupported NGINX configuration to NGF by Application
+  developers in a safe and uncomplicated manner. However, SnippetsTemplates require some up-front work from Cluster
+  operator, meaning it cannot be implemented quickly, in contrast with SnippetsPolicy.
+
+### SnippetsPolicy
+
+Snippets allow inserting NGINX configuration into various NGINX contexts.
+
+#### API
+
+> Field validation rules are intentionally left out of this proposal.
+
+```go
+// SnippetsPolicy is a Direct Attached Policy. It allows inserting NGINX configuration into the generated NGINX
+// config that affects resources referenced by the policy.
+type SnippetsPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the SnippetsPolicy.
+	Spec SnippetsPolicySpec `json:"spec"`
+
+	// Status defines the state of the SnippetsPolicy.
+	Status gatewayv1alpha2.PolicyStatus `json:"status,omitempty"`
+}
+
+// SnippetsPolicyList contains a list of SnippetsPolicies.
+type SnippetsPolicyList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []SnippetsPolicy `json:"items"`
+}
+
+// SnippetsPolicySpec defines the desired state of the SnippetsPolicy.
+type SnippetsPolicySpec struct {
+	// Snippets is a list of NGINX configuration snippets.
+	// There can only be one snippet per context.
```

Comment:
I can't find a reason why it could be useful.
at the same time, let's say we have a SnippetPolicy with snippets for bunch of contexts. If we have only one `http` context snippet vs multiple scattered around, it's gonna be easier to read that snippet

Reply:
Can you add that context somewhere in the doc so we don't forget why we have this limitation?